src/date_type.h

```diff
@@ -63,20 +63,25 @@ static const TimerGameCalendar::Year ORIGINAL_MAX_YEAR  = 2090;
  * @param year the year to get the leap years till.
  * @return the number of leap years.
  */
-#define LEAP_YEARS_TILL(year) ((year) == 0 ? 0 : ((year) - 1) / 4 - ((year) - 1) / 100 + ((year) - 1) / 400 + 1)
+static inline TimerGameCalendar::Year LeapYearsTill(TimerGameCalendar::Year year)
```

Comment:
I do not think that counting the number of leap years should return a `Year`. Having said that, maybe just move this functionality into the `DaysTill` function. And have it as `uint number_of_leap_years = ...`.